### PR TITLE
feat: implement slash command extras

### LINKED ISSUE #301: Slash command extras
### Summary

Make extras for commands work with slash commands too

### What is the feature request for?

disnake.ext.commands

### The Problem

The problem is that extras work for normal commands, but not slash commands.  - having extras for slash commands would allow us to categorize them for help commands. 

https://docs.disnake.dev/en/latest/ext/commands/api.html?highlight=extra#disnake.ext.commands.Command.extras

For example:

fun commands:
/fun minesweeper
/fun rocknsizzors

moderation commands:
/ban
/kick

currently this is not possible due to extras not being supported with slash cmds.

### The Ideal Solution

Making extras work with slash cmds

### The Current Solution

if command.name == fun minesweeper:
category = "fun"
if command.name == fun rocknsizzors:
category = "fun"
if command.name == ban:
category = "moderation"

not ideal :p

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Chromosomologist**
Closes #301.

## Summary

Implement a system similar to `commands.Command.extras` for slash-commands: `InvokableApplicationCommand`, `InvokableUserCommand`, `InvokableMessageCommand`, `SubCommandGroup` and `SubCommand` now all support having extras set on them, typed as a `Dict[str, Any]`.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/ext/commands/base_core.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def __init__(self, func: CommandCallback, *, name: str = None, **kwargs):
         self.qualified_name: str = self.name
         # only an internal feature for now
         self.guild_only: bool = kwargs.get("guild_only", False)
+        self.extras: Dict[str, Any] = kwargs.get("extras") or {}
```
Comment (shiftinv) [inline disnake/ext/commands/base_core.py:116]:
```suggestion
        self.extras: Dict[str, Any] = kwargs.get("extras", {})
```
might be better, the prefix command one works similarly

additionally, the docstring of this type was updated in #470, so the new attribute should be added to it as well

**Review by shiftinv — COMMENTED** [review]
looks good!

The `InteractionBotBase` methods are missing the `extras` parameter, and while it's already handled by `**kwargs`, it would be better make it explicit

File: disnake/ext/commands/base_core.py (hunk 1 above)
Comment (Chromosomologist) [reply disnake/ext/commands/base_core.py:116]: I used or here intentionally as a couple different places pass `extras` to here with default `None`, to keep consistent with the existing spec. For example, `SubCommandGroup.sub_command` had all its kwargs default to `None`. To follow suit, I did the same, but that means None gets explicitly passed to the underlying implementation here, which is why I had to use or (or some other alternative, but that seemed even lengthier, unless you have a better idea).

As for the new attribute, will do in a bit ^^.

**Review by shiftinv — APPROVED** [review]

File: disnake/ext/commands/base_core.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/ext/commands/base_core.py:145]: I found a issue where the user may pass a dict which is edited *later*. This should be documented as unsupported, with the same disclaimer that `Command.extras` has: `This object may be copied by the library.`

Comment (Chromosomologist) [reply disnake/ext/commands/base_core.py:145]: Thanks ^^, disclaimers were added as of 0caa4bf5dcef5e374fef37b09e14fcfa4c1efe47.

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**